src/types.ts

```
@@ -22,7 +22,13 @@ export type RsiProps<T extends string> = {
   // Runs after column matching and on entry change
   tableHook?: TableHook<T>
   // Function called after user finishes the flow. You can return a promise that will be awaited.
-  onSubmit: (mode: ImportMode, data: Result<T>, file: File) => void | Promise<any>
+  onSubmit: (params: {
+    importMode: ImportMode // Ensure ImportMode is imported and defined correctly
+    primaryKeys: string[] // Array of strings representing primary keys
+    data: Result<T> // The data type you are expecting (make sure Result<T> is defined)
+    file: File // File object representing the uploaded file
+  }) => void | Promise<any>
```

Biome

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)